### PR TITLE
Add privacy and security self review questionnaire

### PR DESCRIPTION
**Pull request by alexnj**
cc @domenic

### CODE REVIEW

File: privacy-security-questionnaire.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+## Self-Review Questionnaire: Security and Privacy
+
+| 2.1. What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?
+
+This feature is a Document-Policy header that an origin may choose to include in an HTTP resonse when it has no linked resources in the response, to indicate to the user agent that it may optimize resources required to conduct a speculative parse of the response. This will benefit the user in terms of reducing compute requirements on the client, and improve performance of the experience.
```
Comment (domenic) [inline privacy-security-questionnaire.md]: I don't think this answers the question. This seems like a general description of the feature, instead.

In terms of what information this feature exposes to Web sites or other parties, I think the answer is "none"? Like, look at the sub-questions in https://www.w3.org/TR/security-privacy-questionnaire/#purpose .

File: privacy-security-questionnaire.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+| 2.2. Do features in your specification expose the minimum amount of information necessary to enable their intended uses?
+
+Yes. The feature only exposes the minimum required information as a boolean header, to satisfy the use case.
+
+| 2.3. How do the features in your specification deal with personal information, personally-identifiable information (PII), or information derived from them?
+Personal information is any data about a user (for example, their home address), or information that could be used to identify a user, such as an alias, email address, or identification number.
```
Comment (domenic) [inline privacy-security-questionnaire.md]: I don't think restating the definition of PI is helpful here.

File: privacy-security-questionnaire.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+The feature does not relate to PII or any derived information thereof.
+
+| 2.4. How do the features in your specification deal with sensitive information?
+Personal information is not the only kind of sensitive information. Many other kinds of information may also be sensitive. What is or isn’t sensitive information can vary from person to person or from place to place. Information that would be harmless if known about one person or group of people could be dangerous if known about another person or group. Information about a person that would be harmless in one country might be used in another country to detain, kidnap, or imprison them.
```
Comment (domenic) [inline privacy-security-questionnaire.md]: Oh, I see, the problem is you're copying more from the questionnaire. I don't think that's a good idea, especially without the quote marks.

In general, consider using the template at https://raw.githubusercontent.com/w3ctag/security-questionnaire/main/questionnaire.markdown

File: privacy-security-questionnaire.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+No. The feature in the specification only applies to a single response and does not persist across browsing session.
+
+| 2.6. Do the features in your specification expose information about the underlying platform to origins?
+
+No. The feature does not expose information about the underlying platform.
+
+| 2.7. Does this specification allow an origin to send data to the underlying platform?
+
+The specification only allows an origin to send a signal to the user agent on how to better handle the current response to achieve higher performance for the user experience.
```
Comment (domenic) [inline privacy-security-questionnaire.md]: I think the answer is just "no", if you look at the definition of "underlying platform" in this question.

File: privacy-security-questionnaire.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+| 2.8. Do features in this specification enable access to device sensors?
+
+No.
+
+| 2.9. Do features in this specification enable new script execution/loading mechanisms?
+
+The specification enables a user agent to not attempt speculatively parsing the current response, or loading resources speculatively from it.
```
Comment (domenic) [inline privacy-security-questionnaire.md]: This doesn't answer the question, instead describing the feature.

File: privacy-security-questionnaire.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+No difference.
+
+| 2.15. Does this specification have both "Security Considerations" and "Privacy Considerations" sections?
+
+Feature in the specification does not have security or privacy impacts as it only indicates to the browser that there are no linked resources in a document.
+
+| 2.16. Do features in your specification enable origins to downgrade default security protections?
+
+The Document-Policy in the specification is an optional one, on both sides. The origin may avoid including it, and the browser may choose to ignore it.
```
Comment (domenic) [inline privacy-security-questionnaire.md]: Doesn't answer the question.

File: privacy-security-questionnaire.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+| 2.17. How does your feature handle non-"fully active" documents?
+
+Not applicable.
```
Comment (domenic) [inline privacy-security-questionnaire.md]: Explain in more detail. (Something like: this only impacts initial document processing, so nothing after navigating away interacts with this feature.)

File: privacy-security-questionnaire.md (hunk 2 above)
Comment (alexnj) [reply privacy-security-questionnaire.md]: Ouch. that was a copy paste error. Line 14 was supposed to be the answer :)

File: privacy-security-questionnaire.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+>      Considerations" sections?
+
+No. The feature in the specification does not have any security or privacy impact.
+
+> 17.  Do features in your specification enable origins to downgrade default
+>      security protections?
+
+No.
+
+| 2.17. How does your feature handle non-"fully active" documents?
```
Comment (domenic) [inline privacy-security-questionnaire.md]: Questions 18-21 of the template don't match this.

I guess the difference is https://w3ctag.github.io/security-questionnaire/ (latest version) vs. https://www.w3.org/TR/security-privacy-questionnaire/ (older version). Ugh. And neither perfectly matches the markdown template.

I say stick with the Markdown template, and I'll file an issue about this mismatch.

File: privacy-security-questionnaire.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+No. The feature does not relate to PII or any derived information thereof.
+
+> 04.  How do the features in your specification deal with sensitive information?
+
+No. The feature does not deal with sensitive information.
+
+> 05.  Does data exposed by your specification carry related but distinct
+>      information that may not be obvious to users?
+
+Yes. The specification carries related information about the page, specifically if there are linked resources to be expected in the requested page, as a Document-Policy header in the response header, which will not be obvious to users.
```
Comment (domenic) [inline privacy-security-questionnaire.md]: Reading https://w3ctag.github.io/security-questionnaire/#hidden-data in full, I think "This feature does not allow users to share data with origins" is probably a better answer.

File: privacy-security-questionnaire.md (hunk 8 above)
Comment (alexnj) [reply privacy-security-questionnaire.md]: There is some difference in questions, yes. I updated to stick with the template.

**Review by domenic — APPROVED** [review]

**Pull request merged**